### PR TITLE
Drop Node 14 and require Node 16 or higher

### PR DESCRIPTION
**Pull request by ardatan**

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     with:
       installDependencies: true
       packageManager: pnpm
-      nodeVersion: 18
+      nodeVersion: 20
```
Comment (n1ru4l) [inline .github/workflows/pr.yml]: lets use the LTS version here and in all other sections

Comment (ardatan) [reply .github/workflows/pr.yml]: 👍

**Comment by n1ru4l** [comment]
tsconfig target and lib can be bumped to es2021, see https://github.com/kamilkisiela/graphql-config/blob/master/tsconfig.json#L4-L5

**Comment by ardatan** [comment]
Done @n1ru4l 👍

**Review by dotansimha — APPROVED** [review]

**Pull request merged**